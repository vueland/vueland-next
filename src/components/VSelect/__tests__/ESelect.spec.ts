import { mount } from '@vue/test-utils'
import { ESelect } from '../index'
import { items } from './mocks'
import 'regenerator-runtime'

describe('ESelect', () => {
  const mountFunction = (options = {}) => mount(ESelect as any, {
    ...options,
    attachTo: document.querySelector('.e-app') as HTMLElement
  })

  beforeEach(() => document.body.innerHTML = `<div class="e-app"></div>`)

  it('should mount component and match snapshot', async () => {
    mountFunction({ props: { items, valueKey: 'profession.programmer' } })
    // wait for mounting of selects list
    await new Promise(resolve => setTimeout(resolve))
      .then(() => {
        expect(document.body).toMatchSnapshot()
      })
  })

  it('should test displaying of select list', async () => {
    const wrapper = mountFunction({ props: { items, valueKey: 'name' } })
    // wait for mounting of select list
    await new Promise(resolve => setTimeout(resolve))

    await wrapper.find('.e-input__text-field').trigger('click')
    // wait
    await new Promise(resolve => setTimeout(resolve, 100))

    const menuContent = document.body.querySelector('.e-menu__content')!
    const contentStyle = menuContent.getAttribute('style')!

    expect(contentStyle.includes('display')).toBeFalsy()
  })

  it('should test "readonly" state', async () => {
    const wrapper = mountFunction({
      props: {
        readonly: true
      }
    })

    expect(wrapper.find('.e-select').attributes('class')).toContain('e-select--readonly')

    await wrapper.find('.e-input').trigger('click')
      .then(() => {
        const menuContent = document.body.querySelector('.e-menu__content')!
        const contentStyle = menuContent.getAttribute('style')!

        expect(contentStyle.includes('display: none;')).toBeTruthy()
      })
  })

  it('should test "disabled" state', async () => {
    const wrapper = mountFunction({
      props: {
        disabled: true
      }
    })

    expect(wrapper.find('.e-select').attributes('class')).toContain('e-select--disabled')

    await wrapper.find('.e-input').trigger('click')
      .then(() => {
        const menuContent = document.body.querySelector('.e-menu__content')!
        const contentStyle = menuContent.getAttribute('style')!

        expect(contentStyle.includes('display: none;')).toBeTruthy()
      })
  })
})
