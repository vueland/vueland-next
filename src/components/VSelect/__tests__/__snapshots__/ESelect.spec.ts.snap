// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESelect should mount component and match snapshot 1`] = `
<body>
  <div
    class="e-app"
  >
    <div
      data-v-app=""
    >
      <div
        class="e-input e-input--primary e-input--has-append-icon"
      >
        <div
          class="e-input__label"
        >
          <label
            class="e-label"
          >
            
          </label>
        </div>
        <div
          class="e-input__text-field"
        >
          <!---->
          
          <div
            class="e-select"
          >
            <input
              class="e-select__input dark--text text--graphite-2"
              readonly=""
              type="text"
            />
          </div>
          
          <div
            class="e-input__append-icon"
          >
            <i
              class="e-icon fas fa-chevron-down"
              style="font-size: 16px;"
            />
          </div>
        </div>
        <div
          class="e-input__hints"
        >
          <transition-stub>
            <!---->
          </transition-stub>
        </div>
        <div
          class="e-input__selects"
        >
          
          <div
            class="e-menu"
          />
          <!---->
          <transition-stub />
          
        </div>
      </div>
    </div>
    <div
      class="e-menu__content elevation-10"
      style="top: 0px; left: 0px; z-index: 10; display: none;"
    >
      <div
        class="e-menu__slot"
        style="max-height: 240px; width: 0px;"
      >
        
        <div
          class="e-select-list"
        >
          <div
            class="e-list e-list--active"
            style="background-color: rgb(255, 255, 255); border-color: #ffffff;"
          >
            
            <div
              class="e-list-item"
            >
              <div
                class="e-list-item__title"
              >
                javascript
              </div>
            </div>
            <div
              class="e-list-item"
            >
              <div
                class="e-list-item__title"
              >
                Kotlin
              </div>
            </div>
            <div
              class="e-list-item"
            >
              <div
                class="e-list-item__title"
              >
                Rust
              </div>
            </div>
            
          </div>
        </div>
        
      </div>
    </div>
  </div>
</body>
`;
